### ISSUE TITLE
Install fails when package isn't in "src" folder

### ISSUE THREAD

**Issue by drewcassidy** [scikit-build/scikit-build#686]
Im attempting to port my project from a bespoke build system to scikit-build, but I found that when the source isn't stored in a folder called "src", it fails and tries to install the extension module directly in the root of the virtual environment
<img width="403" alt="Screen Shot 2022-04-03 at 11 56 14 PM" src="https://user-images.githubusercontent.com/3837797/161489514-c0763a30-c424-423f-8f61-4ed8aa93a9fa.png">

my repo as it currently stands is https://github.com/drewcassidy/quicktex/tree/421876ab0f9098cdc21f7c973d8f18b2488a8fc9

I was able to reproduce it with the [scikit_build_example](https://github.com/pybind/scikit_build_example) project by moving "[scikit_build_example](https://github.com/pybind/scikit_build_example/tree/master/src/scikit_build_example)" to the project root and adjusting setup.py accordingly

**Comment by drewcassidy** [comment]
Looks like this may be due to me jumping the gun and using PEP621 setuptools configuration immediately after it was released. I'll test this some more later

**Comment by drewcassidy** [comment]
I put together this fork of scikit_build_example to demonstrate the issue https://github.com/drewcassidy/scikit_build_example

in the process I found another issue: "package_dir={"": "."}, " in setup() causes an error that only exists when using the scikit-build setup() function: `error: package directory '//scikit_build_example' does not exist`. Not sure if this is a bug or not

**Comment by henryiii** [comment]
I'm pretty sure there's a bug lurking that https://github.com/scikit-build/scikit-build/pull/582 was trying to fix, but it's been hard to reproduce. Possibly related?

**Comment by drewcassidy** [comment]
Maybe. I'm not doing an editable install, just `pip install` but the root cause may be the same

**Comment by henryiii** [comment]
The fiddling with the package dir likely would affect more than just editable installs.

We very much do not support PEP 621 configuration yet (see https://github.com/scikit-build/scikit-build/discussions/601 or our wiki for potential plans there) - we don't even support setup.cfg for several values (see #591).

Do you have a commit that makes the "does not exist" error?

**Comment by drewcassidy** [comment]
https://github.com/drewcassidy/scikit_build_example/blob/master/setup.py#L23-L25 See here, uncommenting that line causes the error

**Comment by drewcassidy** [comment]
its also possible this is related to https://github.com/pypa/setuptools/issues/3246, how fun

**Comment by abravalheri** [comment]
Hi @henryiii, is there anything we could facilitate here from `setuptools` point of view?

**Comment by henryiii** [comment]
Not _right_ now - biggest issue is we have a wrapper around `setup` that's a pain to maintain and doesn't intact well with commands. We can't handle setup.cfg, much less pyproject.toml commands, due to that. I think https://iscinumpy.dev/post/scikit-build-proposal/ is likely the best solution, but I don't know if that will happen. I _think_ if we could remove some/much of the wrapper and use the "setup prepare" (forgot what it was called) hook to get the values from setuptools instead of parsing and modifying them ourselves. In fact, the "modifying them" is the part that's currently causing problems...

We need to improve our test suite, I feel like that's the top priority... 🤦

**Comment by drewcassidy** [comment]
Since I don't think I made it very clear: the PEP621 thing is a red herring (though it may also be a dealbreaker depending on what data exactly is read from `setup()`). The forked example repo linked above is not using PEP621 metadata and only differs from the upstream `scikit_build_example` by source location

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
